src/main/java/com/aws/greengrass/clientdevices/auth/iot/registry/CertificateRegistry.java

```diff
@@ -22,10 +25,11 @@
 
 public class CertificateRegistry {
     private static final Logger logger = LogManager.getLogger(CertificateRegistry.class);
-    // holds mapping of certificateHash (SHA-256 hash of certificatePem) to IoT Certificate Id;
+    // holds mapping of certificateHash (SHA-256 hash of certificatePem) to its registry entry;
     // size-bound by default cache size, evicts oldest written entry if the max size is reached
-    private final Map<String, String> registry = Collections.synchronizedMap(
-            new LinkedHashMap<String, String>(RegistryConfig.REGISTRY_CACHE_SIZE, 0.75f, false) {
+    @Getter(AccessLevel.PROTECTED)
```

Comment:
Protected getters always make me raise an eyebrow. Avoid doing this, ESPECIALLY just for unit tests. If you can't initialize your object under test using the public API, then your test is too tightly coupled to the implementation and you should refactor.
Your tests for this class should not break when we replace this map with something else.